pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/immutable/ImmutableSegmentImpl.java

```diff
@@ -173,6 +175,16 @@ private File getValidDocIdsSnapshotFile() {
         V1Constants.VALID_DOC_IDS_SNAPSHOT_FILE_NAME);
   }
 
+  public boolean needReloadSegment(String segmentName, String crc, IndexLoadingConfig indexLoadingConfig)
+      throws Exception {
+    Schema schema = indexLoadingConfig.getSchema();
+    //if re processing or reload is needed on a segment then return true
+    if (_segmentDirectory != null) {
+      return ImmutableSegmentLoader.needPreprocess(_segmentDirectory, indexLoadingConfig, schema);
+    }
+    return false;
```

Comment:
(minor) The null check is redundant
```suggestion
    return ImmutableSegmentLoader.needPreprocess(_segmentDirectory, indexLoadingConfig, indexLoadingConfig.getSchema());
```